functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy

```diff
@@ -303,17 +303,61 @@ abstract class BaseContainer extends Specification implements ClientProvider {
         }
     }
 
+    /**
+     * Deletes the specified project.
+     * This method sends a DELETE request to remove the project with the given name.
+     * If the deletion operation fails, a RuntimeException is thrown.
+     *
+     * @param projectName the name of the project to be deleted. Must not be null.
+     * @throws RuntimeException if the project deletion fails.
+     *         The exception contains a detailed message obtained from the server's response.
+     */
     void deleteProject(String projectName) {
         def response = client.doDelete("/project/${projectName}")
         if (!response.successful) {
             throw new RuntimeException("Failed to delete project: ${response.body().string()}")
         }
     }
 
+    /**
+     * Disables scheduled executions for a specific project and then deletes the project.
+     * This method first makes a PUT request to update the project's configuration,
+     * specifically to disable all scheduled executions. If this operation is successful,
+     * it proceeds to delete the project with a DELETE request. If any of the operations fail,
+     * a RuntimeException is thrown.
+     *
+     * @param projectName the name of the project to be disabled and deleted. Must not be null.
+     * @param body a map containing the configuration to be updated in the project before deletion.
+     *        Specifically, this map should include the necessary properties to disable
+     *        scheduled executions. The exact contents of the map will depend on the client API and
+     *        the project configuration.
+     * @throws RuntimeException if disabling scheduled executions or deleting the project fails.
+     *         The exception contains a detailed message obtained from the server's response.
+     */
+    void disableScheduledAndDeleteProject(String projectName, Map body) {
```

Comment:
method name and description does not match with the actual functionality, the method can be used to set any property value, not just disable schedule.
It would also be useful to have this separated into 2 methods, one to set the properties and delete on a separated mehtod

Reply:
Okay, I have divided it into 2, this method changes the configuration of the project depending on the values you send it, and already exists the delete method.